hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4725,6 +4728,48 @@ func (r *HostedClusterReconciler) isAutoscalingNeeded(ctx context.Context, hclus
 	return false, nil
 }
 
+func (r *HostedClusterReconciler) syncKVLiveMigratableCondition(ctx context.Context, hcluster *hyperv1.HostedCluster) error {
+	if hcluster.Spec.Platform.Type != hyperv1.KubevirtPlatform {
+		return nil
+	}
+	nodePools, err := listNodePools(ctx, r.Client, hcluster.Namespace, hcluster.Name)
+	if err != nil {
+		return fmt.Errorf("failed to get node pools by cluster name for cluster %q: %w", hcluster.Name, err)
+	}
+	var nonMigratableNodePools []string
+	for _, nodePool := range nodePools {
+		lmNpCondition := FindNodePoolStatusCondition(nodePool.Status.Conditions, string(hyperv1.KubeVirtNodesLiveMigratableType))
+		if lmNpCondition != nil && lmNpCondition.Status == corev1.ConditionFalse {
+			nonMigratableNodePools = append(nonMigratableNodePools, nodePool.Name)
+		}
+	}
+
+	var lmHcCondition metav1.Condition
+
+	if len(nonMigratableNodePools) == 0 {
+		lmHcCondition = metav1.Condition{
+			Type:               string(hyperv1.KubeVirtNodesLiveMigratableType),
+			Status:             metav1.ConditionTrue,
+			Reason:             hyperv1.AsExpectedReason,
+			ObservedGeneration: hcluster.Generation,
+		}
+	} else {
+		lmHcCondition = metav1.Condition{
+			Type:   string(hyperv1.KubeVirtNodesLiveMigratableType),
+			Status: metav1.ConditionFalse,
+			Reason: hyperv1.KubeVirtNodesLiveMigratableReason,
+			Message: fmt.Sprintf("Non-live-migratable node(s) have been found on the following NodePools attached to "+
+				"this Hosted Cluster: %s. Consider using ReadWriteMany storage to make the nodes live-migratable and "+
+				"improve cluster stability and downtime",
```

Comment:
I think it would be better if we didn't give any instruction in this condition's reason related to RWX. There are reasons outside of RWX storage that can cause a VM to not be live migratable.

Reply:
the negative condition message has been rephrased.